cmd/kubeapps-apis/server/server.go

```diff
@@ -243,5 +245,41 @@ func gatewayMux() (*runtime.ServeMux, error) {
 		return nil, fmt.Errorf("failed to serve: %v", err)
 	}
 
+	svcRestConfig, err := rest.InClusterConfig()
```

Comment:
Using pod's SA to avoid forcing the user's token to have RBAC on operators icons.

Reply:
I'm not 100% sure, but using such a powerful SA for proxying a k8s API when we are directly injecting unsanitized input from the URL (below), seems like a great attack vector for malicious malformed URLs escalating privileges 😅 .

That said, it is something strictly temporary we should fix near soon, and we clearly discourage enabling operators in production envs, so I'm fine with the change given what you told in the PR comments.

`fmt.Sprintf("/apis/packages.operators.coreos.com/v1/namespaces/%s/packagemanifests/%s/icon", namespace, name)`

Reply:
> but using such a powerful SA for proxying a k8s API when we are directly injecting unsanitized input from the URL

Totally agree on this. It seems that `net/http`'s `ServeMux` does some sanitizing of the input (see `cleanPath` [here](https://go.dev/src/net/http/server.go?s=40985:41006#L2440)), but we are using GRPC's `ServeMux` which apparently doesn't.

We could mimic the `net/http` behavior using `path.Clean` or use an external dependency.